cpu/stm32/kconfigs/Kconfig.clk

```diff
@@ -65,15 +65,68 @@ config CLOCK_PLL_M
 
 config CLOCK_PLL_N
     int "N: PLLIN multiply factor" if CUSTOM_PLL_PARAMS
+    default 120 if BOARD_HAS_HSE && CPU_FAM_F2
+    default 60 if CPU_FAM_F2
+    default 168 if CPU_FAM_F4 && CLOCK_MAX_84MHZ && BOARD_HAS_HSE
+    default 84 if CPU_FAM_F4 && CLOCK_MAX_84MHZ
+    default 96 if CPU_FAM_F4 && CLOCK_MAX_100MHZ && BOARD_HAS_HSE && (MODULE_PERIPH_USBDEV || USEMODULE_PERIPH_USBDEV)
+    default 48 if CPU_FAM_F4 && CLOCK_MAX_100MHZ && (MODULE_PERIPH_USBDEV || USEMODULE_PERIPH_USBDEV)
```

Comment:
I think for now this is OK, but perhaps we could think this again later. It looks to me that an specific clock configuration is a dependency of the usbdev module. I'm not sure yet how to make this in a generic way yet, but the fact that bringing in a module changes the board's clock config seems a bit odd. I'd rather have the user know what's going on, and change the configuration if the module is wanted. What do you think?

Reply:
The problem is about the default clock speed. In RIOT, we try to have the maximum possible clock speed.
By default, some F4 are able to run at 100MHz or 180MHz but this clock speed is not compatible with USB (because a 48MHz clock cannot be generated from the main PLL running at 100MHz/180MHz).
Some CPU provides alternative 48MHz sources (f412/413 for example can use one output of PLL I2S), so in this case this is not an issue, just enable the right PLL I2S output and you get the best of both worlds. But for others (f411, f446), there's no alternative and the main PLL must be changed if the user needs USB.
If no 48MHz clock source can be configured, the build fails.

Previously, F4 capable of running at 100MHz/180MHz were down clocked to 96MHz/168MHz, to be able to work in any situation. But I found this less optimal in most of the cases where USB (or HWRNG) is not needed.

With the current approach, we can guarantee an optimal and working clock speed in any case (with and without USB), without having to change anything.

Reply:
I see, we don't have any check in place to catch this invalid clock configuration when USB is used, right?

Reply:
> we don't have any check in place to catch this invalid clock configuration when USB is used, right?

Yes we do, using preprocessor checks in the clock init code:

https://github.com/RIOT-OS/RIOT/blob/41eea7ab64382f219bba428176b6a617318ebf56/cpu/stm32/stmclk/stmclk_f2f4f7.c#L88-L93

Reply:
Ah great, thanks!